plotter/general_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"bytes"
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/vg"
+)
+
+var generateTestData = flag.Bool("regen", false, "Uses the current state to regenerate the test data.")
+
+// checkPlot checks a generated plot against a previously created reference.
+// If generateTestData = true, it regereates the reference.
+func checkPlot(ExampleFunc func(), t *testing.T, paths ...string) {
```

Comment:
I think it would be nicer to take a list of basenames rather than paths - the path base can be Chdir'd to in lieu.

As an aside, does this behave correctly when invoked from somewhere other than `$GOPAPTH/.../gonum/plot`?